docs/core/tools/extensibility.md

```diff
@@ -1,63 +1,70 @@
+
 of the context of the project (for example, outside of the directory that contains the project) will fail as the command will 
 not be able to be found.
 
 These tools are perfect for build servers, since nothing outside of the project file is needed. The build process 
 runs restore for the project it builds and tools will be available. Language projects, such as F#, are also in this 
-category; after all, each project can only be written in one specific language. 
+category since each project can only be written in one specific language. 
 
 Finally, this extensibility model provides support for creation of tools that need access to the built output of the 
 project. For instance, various Razor view tools in [ASP.NET](https://www.asp.net/) MVC applications fall into this 
 category. 
 
 ### Consuming per-project tools
-Consuming these tools requires you to add a `<DotNetCliToolReference>` element for each tool you want to use to your project file. Inside the `<DotNetCliToolReference>` element, you reference the package in which the tool resides and you specify the version you need. After running `dotnet restore`, the tool and its dependencies are restored. 
+Consuming these tools requires you to add a `<DotNetCliToolReference>` element to your project file for each tool you want to use. Inside the `<DotNetCliToolReference>` element, you reference the package in which the tool resides and specify the version you need. After running [`dotnet restore`](dotnet-restore.md), the tool and its dependencies are restored. 
 
 For tools that need to load the build output of the project for execution, there is usually another dependency which is 
-listed under the regular dependencies in the project file. Since CLI uses MSBuild as its build engine, it is recommended that these parts of the tool be written as custom MSBuild targets and tasks since that way they can take part in the overall build process. Also, they can get any and all data easily that is produced via the build, for example the location of the output files, the current configuration being built etc. All of this information becomes a set of MSBuild properties that can be read from any target. We will see how to add a custom target using NuGet later in this document. 
+listed under the regular dependencies in the project file. Since CLI uses MSBuild as its build engine, it's recommended that these parts of the tool are written as custom MSBuild [targets](https://docs.microsoft.com/visualstudio/msbuild/msbuild-targets) and [tasks](https://docs.microsoft.com/visualstudio/msbuild/msbuild-tasks) since that way they can take part in the overall build process. Also, they can get any and all data easily that is produced via the build, such as the location of the output files, the current configuration being built, etc. All this information becomes a set of MSBuild properties that can be read from any target. You can see how to add a custom target using NuGet later in this document. 
```

Comment:
it's recommended that --> we recommend that 
are written --> be written
comma before since
that way they can -->they can then